src/kvstore/comm.h

```diff
@@ -58,7 +76,10 @@ class Comm {
    */
   virtual void Broadcast(
       int key, const NDArray& src,
-      const std::vector<NDArray*> dst, int priority) = 0;
+      const std::vector<NDArray> dst, int priority) = 0;
+
```

Comment:
Could you add brief comments for these two methods? Are they only for nccl? Do we want to declare it only when MXNET_USE_NCCL is set?